subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3905,7 +3905,7 @@ static void socket_receive_loop(void)
 		 * FIXME: Currently we timeout and restart poll in case fds
 		 *        were modified.
 		 */
-		if (poll(sock_fds, sock_nfds, ENGINE_UPDATE_INTERVAL) < 0) {
+		if ((errno = poll(sock_fds, sock_nfds, ENGINE_UPDATE_INTERVAL)) < 0) {
```

Comment:
Isn't poll() returning always -1, so it does not make sense to assing errno that value.

Reply:
Oh, I just noticed you are right, poll does return -1 when it gets error, and 0 for time out. I thought -EAGAIN or -ETIMEDOUT might return when poll times out.
So this way we should not need to report the error number with LOG_ERR I think.